cli/azd/.vscode/launch.json

```diff
@@ -0,0 +1,14 @@
+{
+    // If you set `AZD_DEBUG=true` in your environment, `azd` will pause early in start up and allow you to attach
+    // to it. Use the Attach to Process configuration and pick the corresponding `azd` process.
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Attach to Process",
+            "type": "go",
+            "request": "attach",
+            "mode": "local",
+            "processId": "${command:pickGoProcess}"
+        }
```

Comment:
Since we're already adding a `launch.json`, how about if we also added one for launching `azd` directly? And modify the doc to indicate it.

Here's my proposed launch config:

```json
{
            "name": "dev-azd (launch)",
            "type": "go",
            "request": "launch",
            "mode": "debug",
            "program": "${workspaceFolder}",
            "console": "integratedTerminal",
            "args": [
                // Modify below to the desired launch args
                "init",
                "--debug"
            ],
            // Modify below to the desired azd project path
            "cwd": "${workspaceFolder}/test/functional/testdata/samples/storage"
        }
```

Note that we can make `args` to be possible to be prompted when the go debugger extension supports this.

Reply:
IMO - devs would need to manually modify the launch config for each desired cmd / args combo.  I find it much easier to just use the `AZD_DEBUG=true` env var and just manually attach to the process in the current `launch.json`.

Reply:
@wbreza That's correct, but even if we didn't include the config as a sample -- that is still needed. You'll still need launch debug for debugging early execution logic. 

Also, I did some research in the area, and it looks like if the go debugger added support for a single prompt string https://github.com/golang/vscode-go/issues/2621, we could avoid the manual edits.

Reply:
>  manually modify the launch config for each desired cmd / args combo

the cmd/args combo needs to be written any way.  It's not like we can save us from typing it.  What we can save is the `pick process` to attach.

Here's my proposal (a little addition to Wei's):
Note the `env` and the `console` as `integratedTerminal` (to allow I/O interaction during debug)

```
{
            "name": "init",
            "type": "go",
            "request": "launch",
            "mode": "debug",
            "program": "${workspaceFolder}",
            "args": [
                "init"
            ],
            "console": "integratedTerminal",
            "cwd": "~/workspace/build/p",
            "env": {
            }
        },
```